php/libraries/ParentIDGenerator.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types=1);
+/**
+ * This file contains a class used to generate ParentIDs for use in LORIS.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  LORIS
+ * @author   Shen Wang <wangshen.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\StudyEntities\Candidate\ParentIDGenerator;
+/**
+ * This class defines functions used to generate valid ParentID identifiers in
+ * LORIS.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  LORIS
+ * @author   Shen Wang <wangshen.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class ParentIDGenerator extends \CandIDGenerator
```

Comment:
i don't think the ParentIDGenerator should be extending the CandIDGenerator and call its generate() method because that method checks for whether a certain randomly generated CandID already exists in the databases. The ParentID is separate to the CandID and should not be dependent on it at all